collections/ancestrynew.js

```diff
@@ -79,116 +79,97 @@ registerCollection({
 
 var ancestrymrglist = [];
 
-function parseAncestryNew(htmlstring, familymembers, relation) {
+async function readPersonCard(htmlstring) {
+    var foundScript = $("script:contains('var PersonCard')", htmlstring).text();
+    
+    return new Promise((resolve, reject) => {
+        // TODO assign the return of this to the PersonCard object
+        if (foundScript) {
+            // truncate the end
+            foundScript = foundScript.substring(0, foundScript.indexOf("};") + 2) + "PersonCard;";
+            chrome.runtime.sendMessage({
+                action: "eval",
+                variable: foundScript
+            }).then((resp) => {
+                // TODO - need to figure out how to use this
+                resolve(resp);
+            });
+        } else {
+            reject("PersonCard not found");
+        }
+    });
+
+}
+
+async function parseAncestryNew(htmlstring, familymembers, relation) {
     relation = relation || "";
     if (!exists(htmlstring)) {
         return "";
     }
-    var parsed = $(htmlstring.replace(/<img/ig, "<gmi"));
-    var par = parsed.find("#personCard");
-    var focusperson = par.find(".userCardTitle").text();
-    var focusdaterange = par.find(".userCardSubTitle").text().replace("&ndash;", " - ");
+    // TODO - figure out where the parsing issues are
+    var personCard = await readPersonCard(htmlstring);
+
+    // get personFacts
+    var pfStart = htmlstring.indexOf("researchData = ");
+    var pfEnd = htmlstring.indexOf("};", pfStart);
+
+    var personFacts = JSON.parse(htmlstring.substring(pfStart + 14, pfEnd + 1));
```

Comment:
Be careful with these numeric increments, if Ancestry would decide to add another attribute to their HTML tag (i.e. script tag), you're toast and it's broken.
It's much better, IMO, to find the first `var x = {` and start from the `{`